### PR TITLE
[Tooling] `yarn splash` (beta)

### PR DESCRIPTION
**Pull request by kaelig**
### First iteration

Screen Shot 2019-09-09 at 11 33 26 PM

### Second iteration (with real data)

Screen Shot 2019-09-10 at 11 02 27 PM

#### No changes detected
Screen Shot 2019-09-11 at 11 27 12 PM

#### Loading
Screen Shot 2019-09-11 at 11 26 53 PM

#### Changes detected
Screen Shot 2019-09-11 at 11 26 59 PM

### CODE REVIEW

File: scripts/observer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import tinycolor from 'tinycolor2';
+import tokens from '@shopify/polaris-tokens';
+import {Box, Text, render, Color} from 'ink';
+
+const data = [
```
Comment (kaelig) [inline scripts/observer/index.tsx]: @amrocha Here's the kind of data I'd expect from the other side, but pretty much anything less structured also would work, as long as the same amount of raw information is available.

File: scripts/observer/index.tsx [hunk 2]
```diff
@@ -1,5 +1,13 @@
 import React from 'react';
 import {Box, Text, Color, render} from 'ink';
+import {getGitStagedFiles, getDependencies} from '../../treebuilder';
+
+(async function() {
+  const stagedFiles = await getGitStagedFiles('src/');
+  console.log(stagedFiles);
+  const data = await getDependencies(/* how do I use this function? */);
```
Comment (kaelig) [inline scripts/observer/index.tsx]: @amrocha I couldn't figure out what was expected to be passed here. A little nudge would be appreciated!

File: treebuilder.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+      node.dependsOn = dependencies;
+    }
+
+    graph[path.resolve(node.fileName)] = node;
+    return node;
+  }
+
+  function skipIndexFile(fileName) {
+    if (/(components\/)(\w*\/)?(index.ts)/.test(fileName)) {
```
Comment (kaelig) [inline treebuilder.ts]: @amrocha This seems to cause mischief in some cases, can you have a look?

### Steps to reproduce

1. Make a change to `src/components/Modal/components/Footer/Footer.tsx`
2. Run `yarn observer`

### Expected

```
🧩  src/components/Modal/components/Footer/Footer.tsx
    Component name         File potentially affected (total: 5)
    <Frame />              src/components/Frame/Frame.tsx
    <Frame />              src/components/Frame/components/ContextualSaveBar/ContextualSaveBar.tsx
    <Frame />              src/components/Frame/components/ContextualSaveBar/components/DiscardConfirmationModal/DiscardConfirmationModal.tsx
    <Modal />              src/components/Modal/Modal.tsx
    <Toast />              src/components/Toast/Toast.tsx
```

### Actual

```
🧩  src/components/Modal/components/Footer/Footer.tsx
    Component name         File potentially affected (total: 0)
```

Comment (amrocha) [reply treebuilder.ts]: Oh interesting, taking a look

File: treebuilder.ts [hunk 4]
```diff
@@ -103,7 +103,7 @@ function findDependencies(fileName) {
 
 export function getGitStagedFiles(scope = '') {
   return new Promise((resolve, reject) => {
-    cmd.get('git status --no-renames -s', (err, data, stderr) => {
+    cmd.get('git status --porcelain', (err, data, stderr) => {
```
Comment (kaelig) [inline treebuilder.ts]: This flag seems to be the way to go when parsing git status. Not sure it's super important in this very case, but I thought we might as well use it for what it's designed.

**Pull request merged**